packages/medical-domain/src/domains/repositories/statistics/glycemia-statistics.service.ts

```diff
@@ -0,0 +1,278 @@
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter)) as Bg[]
+  const totalBgValue = filteredBg.reduce((sum, bgData) => sum + bgData.value, 0)
+  const numberOfBg = filteredBg.length
+  return {
+    averageGlucose: totalBgValue / numberOfBg,
+    total: numberOfBg
+  }
+}
+
+function getCoefficientOfVariationData(bgData: Cbg[] | Smbg[], dateFilter: DateFilter): CoefficientOfVariationStatistics {
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const average = getAverageGlucoseData(filteredBg, dateFilter)
+  const noDataStatistics = {
+    insufficientData: true,
+    coefficientOfVariation: Number.NaN,
+    total: average.total
+  }
+  if (average.total < 3) {
+    return noDataStatistics
+  }
+
+  const initialAcc: Record<string, number[]> = {}
+  const bgValueByDay = filteredBg.reduce((current, bgData) => {
+    if (current[bgData.localDate] === undefined) {
+      current[bgData.localDate] = [bgData.value]
+      return { ...current }
+    }
+    current[bgData.localDate].push(bgData.value)
+    return { ...current }
+  }, initialAcc)
+
+  const coefficientOfVariationByDate = Object.keys(bgValueByDay).reduce((current: number[], localDate): number[] => {
+    const numBg = bgValueByDay[localDate].length
+    if (numBg >= 3) {
+      const avgGlucose = meanValues(bgValueByDay[localDate])
+      const squaredDiffs = bgValueByDay[localDate].map((bgValue: number) => (bgValue - avgGlucose) ** 2)
+      const sumSquaredDiffs = sumValues(squaredDiffs)
+      const standardDeviation = Math.sqrt(sumSquaredDiffs / (numBg - 1))
+      return [...current, standardDeviation / avgGlucose * 100]
+    }
+    return [...current]
+  }, [])
+  if (coefficientOfVariationByDate.length === 0) {
+    return noDataStatistics
+  }
+  return {
+    insufficientData: false,
+    coefficientOfVariation: meanValues(coefficientOfVariationByDate),
+    total: average.total
+  }
+}
+
+function getGlucoseManagementIndicatorData(cbgData: Cbg[], bgUnit: BgUnit, dateFilter: DateFilter): GlucoseManagementIndicatoStatistics {
+  const insufficientData = {
+    glucoseManagementIndicator: Number.NaN,
+    total: Number.NaN,
+    insufficientData: true
+  }
+  if (diffDays(dateFilter.start, dateFilter.end) < 14) {
+    return insufficientData
+  }
+
+  const filteredCbg = CbgService.filterOnDate(cbgData, dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const totalCbgDuration = getCgmTotalDuration(filteredCbg)
+  if (totalCbgDuration < 14 * MS_IN_DAY * 0.7) {
+    return insufficientData
+  }
+
+  const { averageGlucose, total } = getAverageGlucoseData(filteredCbg, dateFilter)
+  const averageGlucoseMgdl = bgUnit === MGDL_UNITS ? averageGlucose : convertBG(averageGlucose, MMOLL_UNITS)
+
+  return {
+    glucoseManagementIndicator: (3.31 + 0.02392 * averageGlucoseMgdl),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Do you know what these magic numbers are? If yes, could you put them in a const with a name describing what they represent?

Reply:
No Idea of what these values mean... And I can't find any meaning... 
https://www.jaeb.org/gmi/